pkg/api/controller.go

```diff
@@ -373,6 +364,117 @@ func (c *Controller) StartBackgroundTasks(reloadCtx context.Context) {
 	}
 
 	if c.Config.Extensions != nil {
-		ext.EnableScrubExtension(c.Config, c.StoreController, c.Log)
+		ext.EnableScrubExtension(c.Config, c.Log, false, nil, "")
+	}
+
+	c.StartPeriodicTasks()
+}
+
+func (c *Controller) StartPeriodicTasks() {
+	// start periodic gc and/or scrub for DefaultStore
+	go StartPeriodicTasksForImageStore(c.StoreController.DefaultStore, c.Config.Storage.GC, c.Config.Storage.GCInterval,
+		c.Config.Extensions, c.Log)
+
+	if c.Config.Storage.SubPaths != nil {
+		for route, storageConfig := range c.Config.Storage.SubPaths {
+			// Enable running garbage-collect or/and scrub periodically for subImageStore
+			go StartPeriodicTasksForImageStore(c.StoreController.SubStore[route], storageConfig.GC, storageConfig.GCInterval,
+				c.Config.Extensions, c.Log)
+		}
+	}
+}
+
+func StartPeriodicTasksForImageStore(imageStore storage.ImageStore, configGC bool, configGCInterval time.Duration,
+	extensions *extconf.ExtensionConfig, log log.Logger,
+) {
+	var interval time.Duration
+
+	scrubInterval := time.Duration(0)
+	gcInterval := time.Duration(0)
+
+	gc := false
+	scrub := false
+
+	if configGC && configGCInterval != 0 {
+		gcInterval = configGCInterval
+		gc = true
+	}
+
+	if extensions != nil && extensions.Scrub != nil && extensions.Scrub.Interval != 0 {
+		scrubInterval = extensions.Scrub.Interval
+		scrub = true
+	}
+
+	switch {
```

Comment:
This looks should be a routine called func minPeriodicInterval() instead of this unrolled switch